src/lib/users/inactive/inactive-users-controller.ts

```diff
@@ -0,0 +1,102 @@
+import Controller from '../../routes/controller';
+import { ADMIN, IUnleashConfig, IUnleashServices } from '../../types';
+import { Logger } from '../../logger';
+import { InactiveUsersService } from './inactive-users-service';
+import {
+    createRequestSchema,
+    createResponseSchema,
+    emptyResponse,
+    getStandardResponses,
+    IdsSchema,
+    inactiveUsersSchema,
+    InactiveUsersSchema,
+} from '../../openapi';
+import { IAuthRequest } from '../../routes/unleash-types';
+import { Response } from 'express';
+import { OpenApiService } from '../../services';
+export class InactiveUsersController extends Controller {
+    private readonly logger: Logger;
+
+    private inactiveUsersService: InactiveUsersService;
+
+    private openApiService: OpenApiService;
+    constructor(
+        config: IUnleashConfig,
+        {
+            inactiveUsersService,
+            openApiService,
+        }: Pick<IUnleashServices, 'inactiveUsersService' | 'openApiService'>,
+    ) {
+        super(config);
+        this.logger = config.getLogger(
+            'user/inactive/inactive-users-controller.ts',
+        );
+        this.inactiveUsersService = inactiveUsersService;
+        this.openApiService = openApiService;
+
+        this.route({
+            method: 'get',
+            path: '',
+            handler: this.getInactiveUsers,
+            permission: ADMIN,
+            middleware: [
+                openApiService.validPath({
+                    operationId: 'getInactiveUsers',
+                    summary: 'Gets inactive users',
+                    description:
+                        'Gets all inactive users. An inactive user is a user that has not logged in in the last 180 days',
+                    tags: ['Users'],
+                    responses: {
+                        200: createResponseSchema('inactiveUsersSchema'),
+                    },
+                }),
+            ],
+        });
+        this.route({
+            method: 'post',
+            path: '/delete',
+            handler: this.deleteInactiveUsers,
+            permission: ADMIN,
+            middleware: [
+                openApiService.validPath({
+                    operationId: 'deleteInactiveUsers',
+                    summary: 'Deletes inactive users',
+                    description:
+                        'Deletes all inactive users. An inactive user is a user that has not logged in in the last 180 days',
```

Comment:
Maybe `180` should be a constant somewhere, in case we decide to later change it.

Reply:
Added as a constant to the create-inactive-users-service.ts file for now. We can move it around later. Will update the schema as well.